pkg/registry/apis/provisioning/resources/idmapper_test.go

```diff
@@ -0,0 +1,22 @@
+package resources
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestIDMapper(t *testing.T) {
+	obj := &unstructured.Unstructured{}
+
+	dashName, folderName := NamesFromHashedRepoPath("xyz", "path/to/folder/dashboard.json", obj)
+	require.Equal(t, "dashboard-B60LqDCLR6FC", dashName)
+	require.Equal(t, "folder-aObw9Q2jRwTI", folderName)
+
+	name1, f1 := NamesFromHashedRepoPath("xyz", "path/to/folder.json", obj)
```

Comment:
to support setting folder title/labels etc, we can have a yaml|json file with the same prefix as a folder represent a folder?

Reply:
I think this is the simplest, best idea to solve the problem. It'd be nice not lock us into this solution, though, as we could support metadata on the folder itself (e.g. in S3) as well...